packages/database/src/core/SyncTree.ts

```diff
@@ -550,6 +563,36 @@ export function syncTreeAddEventRegistration(
     syncTree.tagToQueryMap.set(tag, queryKey);
   }
   const writesCache = writeTreeChildWrites(syncTree.pendingWriteTree_, path);
+  // TODO: break this down so you do the minimal amount
```

Comment:
It looks like you added this TODO? Could you either clarify this a bit more or get rid of ti?